hadoop/src/main/java/org/carbondata/hadoop/util/ObjectSerializationUtil.java

```
@@ -33,18 +37,29 @@ public static String convertObjectToString(Object obj) throws IOException {
       oos = new ObjectOutputStream(gos);
       oos.writeObject(obj);
     } finally {
-      oos.close();
-      gos.close();
-      baos.close();
+      try {
+        if (oos != null) {
+          oos.close();
+        }
+        if (gos != null) {
+          gos.close();
+        }
+        if (baos != null) {
+          baos.close();
+        }
+      } catch (IOException e) {
+        LOG.error(e);
```

Throwing the exception in finally block is not recommendable as it confuses the user and masks the actual exceptions. In two cases the flow comes to finally block
 First case is after successful completion of flow in try block and reaches finally block , in this case even though the exception the thrown while closing the connections cannot effect the flow, so we just log it.
Second case is something abnormal happened in try block flow and throws exception, so before throwing exception it goes to finally block to close the connections. In this case if we throw exception from finally block , it masks the actual exception.  That is why throwing exception in finally block is not recommended.